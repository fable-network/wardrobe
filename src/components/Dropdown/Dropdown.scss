//
// Dropdown
//
@import '../../globals/scss/colors';
@import '../../globals/scss/typography';
@import '../../globals/scss/variables';
@import '../../globals/scss/import-once';

// Import this only once
@include exports('dropdown') {
  .#{$prefix}--dropdown {

    &__button {
      display: inline-block;
      background-color: #ffffff;
      border: solid 1px #9b9b9b;
      font-family: inherit;
      font-size: 16px;
      line-height: 1;
      color: #313233;
      padding: 6px 10px;

      > .ft-icon {
        margin-left: 0.4em;
        transition-duration: 100ms;

        // Rotate if dropdown is open
        .#{$prefix}--dropdown--is-open & {
          transform: rotateX(-180deg);
          transition: all .15s ease-in-out;
        }
      }
    }
  }
}
