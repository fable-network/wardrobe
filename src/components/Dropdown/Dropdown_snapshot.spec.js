/* eslint-disable */
// This file was automatically generated
import React from 'react';
import 'jest-styled-components';
import Dropdown from './Dropdown';


jest.mock('../../charts/HighChart', () => () => <high-charts />);

describe('Dropdown Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = mount(
        <Dropdown label="Don't push me">
          <Dropdown.Item>
            <nobr>...cause I'm close to the edge</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>I'm trying not to lose my head</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>It's like a jungle sometimes</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>It makes me wonder how I keep from goin' under</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>-- Grandmaster Flash</nobr>
          </Dropdown.Item>
        </Dropdown>
      );
      expect(wrapper1).toMatchSnapshot();

      const wrapper2 = mount(
        <Dropdown label="Don't push me" size="large">
          <Dropdown.Item>
            <nobr>...cause I'm close to the edge</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>I'm trying not to lose my head</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>It's like a jungle sometimes</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>It makes me wonder how I keep from goin' under</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>-- Grandmaster Flash</nobr>
          </Dropdown.Item>
        </Dropdown>
      );
      expect(wrapper2).toMatchSnapshot();

      const wrapper3 = mount(
        <Dropdown label="Don't push me" persist>
          <Dropdown.Item onSelect={() => alert('Clicked 1')}>
            <nobr>...cause I'm close to the edge</nobr>
          </Dropdown.Item>
          <Dropdown.Item onSelect={() => alert('Clicked 2')}>
            <nobr>I'm trying not to lose my head</nobr>
          </Dropdown.Item>
          <Dropdown.Item onSelect={() => alert('Clicked 3')}>
            <nobr>It's like a jungle sometimes</nobr>
          </Dropdown.Item>
          <Dropdown.Item onSelect={() => alert('Clicked 4')}>
            <nobr>It makes me wonder how I keep from goin' under</nobr>
          </Dropdown.Item>
          <Dropdown.Item onSelect={() => alert("Don't push me")}>
            <nobr>-- Grandmaster Flash</nobr>
          </Dropdown.Item>
        </Dropdown>
      );
      expect(wrapper3).toMatchSnapshot();

      const wrapper4 = mount(
        <Dropdown label="Don't push me" preventOutOfBounds>
          <Dropdown.Item>
            <nobr>...cause I'm close to the edge</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>I'm trying not to lose my head</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>It's like a jungle sometimes</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>It makes me wonder how I keep from goin' under</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>-- Grandmaster Flash</nobr>
          </Dropdown.Item>
        </Dropdown>
      );
      expect(wrapper4).toMatchSnapshot();

      const wrapper5 = mount(
        <Dropdown label="Don't push me" preventOutOfBounds={false}>
          <Dropdown.Item>
            <nobr>...cause I'm close to the edge</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>I'm trying not to lose my head</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>It's like a jungle sometimes</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>It makes me wonder how I keep from goin' under</nobr>
          </Dropdown.Item>
          <Dropdown.Item>
            <nobr>-- Grandmaster Flash</nobr>
          </Dropdown.Item>
        </Dropdown>
      );
      expect(wrapper5).toMatchSnapshot();

      const wrapper6 = mount(
        <Dropdown label="Push Me" persist>
          <nobr>You can place any content you prefer here.</nobr>
        </Dropdown>
      );
      expect(wrapper6).toMatchSnapshot();

      const wrapper7 = mount(
        <Dropdown invalid label="Don't push me">
          <Dropdown.Item>
            <nobr>...cause I'm close to the edge</nobr>
          </Dropdown.Item>
        </Dropdown>
      );
      expect(wrapper7).toMatchSnapshot();

      const wrapper8 = mount(
        <Dropdown selected={true} label="Push Me">
          <Dropdown.Item>Check that checkmark</Dropdown.Item>
        </Dropdown>
      );
      expect(wrapper8).toMatchSnapshot();

      const wrapper9 = mount(
        <Dropdown loading={true} label="Something is happening">
          <Dropdown.Item>Hold on I'm loading</Dropdown.Item>
        </Dropdown>
      );
      expect(wrapper9).toMatchSnapshot();

      const wrapper10 = mount(
        <Dropdown disabled={true}>
          <Dropdown.Item>I'll never be seen</Dropdown.Item>
        </Dropdown>
      );
      expect(wrapper10).toMatchSnapshot();

      const wrapper11 = mount(
        <Dropdown label={<h3>h3 label</h3>}>
          <Dropdown.Item>Hey there</Dropdown.Item>
        </Dropdown>
      );
      expect(wrapper11).toMatchSnapshot();

      const wrapper12 = mount(
        <Dropdown open={true}>
          <Dropdown.Item>
            <nobr>You can't close me</nobr>
          </Dropdown.Item>
        </Dropdown>
      );
      expect(wrapper12).toMatchSnapshot();

      const wrapper13 = mount(
        <Dropdown open={false} label="You can't open me">
          <Dropdown.Item>
            <nobr>You will never see me</nobr>
          </Dropdown.Item>
        </Dropdown>
      );
      expect(wrapper13).toMatchSnapshot();

      const wrapper14 = mount(
        <Dropdown label="Don't push me" fluid={true}>
          <Dropdown.Item onSelect={() => alert('Clicked 1')}>
            <nobr>...cause I'm close to the edge</nobr>
          </Dropdown.Item>
          <Dropdown.Item onSelect={() => alert('Clicked 2')}>
            <nobr>I'm trying not to lose my head</nobr>
          </Dropdown.Item>
          <Dropdown.Item onSelect={() => alert('Clicked 3')}>
            <nobr>It's like a jungle sometimes</nobr>
          </Dropdown.Item>
          <Dropdown.Item onSelect={() => alert('Clicked 4')}>
            <nobr>It makes me wonder how I keep from goin' under</nobr>
          </Dropdown.Item>
          <Dropdown.Item onSelect={() => alert("Don't push me")}>
            <nobr>-- Grandmaster Flash</nobr>
          </Dropdown.Item>
        </Dropdown>
      );
      expect(wrapper14).toMatchSnapshot();

    });
  });
});
