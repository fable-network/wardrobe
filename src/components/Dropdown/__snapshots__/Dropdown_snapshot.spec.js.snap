// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component snapshots should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    className={null}
    disabled={false}
    isSelected={false}
    label="Don't push me..."
  >
     
    <DropdownItem
      onClick={[Function]}
    >
       
      <nobr>
        ...cause I'm close to the edge
      </nobr>
       
    </DropdownItem>
     
    <DropdownItem
      onClick={[Function]}
    >
       
      <nobr>
        I'm trying not to lose my head
      </nobr>
       
    </DropdownItem>
     
    <DropdownItem
      onClick={[Function]}
    >
       
      <nobr>
        It's like a jungle sometimes
      </nobr>
       
    </DropdownItem>
     
    <DropdownItem
      onClick={[Function]}
    >
       
      <nobr>
        It makes me wonder how I keep from goin' under
      </nobr>
       
    </DropdownItem>
     
    <DropdownItem
      onClick={[Function]}
    >
       
      <nobr>
        -- Grandmaster Flash
      </nobr>
       
    </DropdownItem>
  </Dropdown>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Dropdown__DropdownButton
          disabled={false}
          onClick={[Function]}
        >
          Don't push me...
          <Dropdown__ToggleIcon
            color="#313233"
            height={9}
            name="caret-down"
            open={false}
            selected={false}
            width={16}
          />
        </Dropdown__DropdownButton>,
        <Dropdown__DropdownPanel
          innerRef={[Function]}
          open={false}
        >
           
          <DropdownItem
            onClick={[Function]}
          >
             
            <nobr>
              ...cause I'm close to the edge
            </nobr>
             
          </DropdownItem>
           
          <DropdownItem
            onClick={[Function]}
          >
             
            <nobr>
              I'm trying not to lose my head
            </nobr>
             
          </DropdownItem>
           
          <DropdownItem
            onClick={[Function]}
          >
             
            <nobr>
              It's like a jungle sometimes
            </nobr>
             
          </DropdownItem>
           
          <DropdownItem
            onClick={[Function]}
          >
             
            <nobr>
              It makes me wonder how I keep from goin' under
            </nobr>
             
          </DropdownItem>
           
          <DropdownItem
            onClick={[Function]}
          >
             
            <nobr>
              -- Grandmaster Flash
            </nobr>
             
          </DropdownItem>
        </Dropdown__DropdownPanel>,
      ],
      "className": null,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Don't push me...",
            <Dropdown__ToggleIcon
              color="#313233"
              height={9}
              name="caret-down"
              open={false}
              selected={false}
              width={16}
            />,
          ],
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Don't push me...",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "#313233",
              "height": 9,
              "name": "caret-down",
              "open": false,
              "selected": false,
              "width": 16,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            " ",
            <DropdownItem
              onClick={[Function]}
            >
               
              <nobr>
                ...cause I'm close to the edge
              </nobr>
               
            </DropdownItem>,
            " ",
            <DropdownItem
              onClick={[Function]}
            >
               
              <nobr>
                I'm trying not to lose my head
              </nobr>
               
            </DropdownItem>,
            " ",
            <DropdownItem
              onClick={[Function]}
            >
               
              <nobr>
                It's like a jungle sometimes
              </nobr>
               
            </DropdownItem>,
            " ",
            <DropdownItem
              onClick={[Function]}
            >
               
              <nobr>
                It makes me wonder how I keep from goin' under
              </nobr>
               
            </DropdownItem>,
            " ",
            <DropdownItem
              onClick={[Function]}
            >
               
              <nobr>
                -- Grandmaster Flash
              </nobr>
               
            </DropdownItem>,
          ],
          "innerRef": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": Array [
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                " ",
                <nobr>
                  ...cause I'm close to the edge
                </nobr>,
                " ",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "...cause I'm close to the edge",
                },
                "ref": null,
                "rendered": "...cause I'm close to the edge",
                "type": "nobr",
              },
              " ",
            ],
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                " ",
                <nobr>
                  I'm trying not to lose my head
                </nobr>,
                " ",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "I'm trying not to lose my head",
                },
                "ref": null,
                "rendered": "I'm trying not to lose my head",
                "type": "nobr",
              },
              " ",
            ],
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                " ",
                <nobr>
                  It's like a jungle sometimes
                </nobr>,
                " ",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "It's like a jungle sometimes",
                },
                "ref": null,
                "rendered": "It's like a jungle sometimes",
                "type": "nobr",
              },
              " ",
            ],
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                " ",
                <nobr>
                  It makes me wonder how I keep from goin' under
                </nobr>,
                " ",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "It makes me wonder how I keep from goin' under",
                },
                "ref": null,
                "rendered": "It makes me wonder how I keep from goin' under",
                "type": "nobr",
              },
              " ",
            ],
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                " ",
                <nobr>
                  -- Grandmaster Flash
                </nobr>,
                " ",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "-- Grandmaster Flash",
                },
                "ref": null,
                "rendered": "-- Grandmaster Flash",
                "type": "nobr",
              },
              " ",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Dropdown__DropdownButton
            disabled={false}
            onClick={[Function]}
          >
            Don't push me...
            <Dropdown__ToggleIcon
              color="#313233"
              height={9}
              name="caret-down"
              open={false}
              selected={false}
              width={16}
            />
          </Dropdown__DropdownButton>,
          <Dropdown__DropdownPanel
            innerRef={[Function]}
            open={false}
          >
             
            <DropdownItem
              onClick={[Function]}
            >
               
              <nobr>
                ...cause I'm close to the edge
              </nobr>
               
            </DropdownItem>
             
            <DropdownItem
              onClick={[Function]}
            >
               
              <nobr>
                I'm trying not to lose my head
              </nobr>
               
            </DropdownItem>
             
            <DropdownItem
              onClick={[Function]}
            >
               
              <nobr>
                It's like a jungle sometimes
              </nobr>
               
            </DropdownItem>
             
            <DropdownItem
              onClick={[Function]}
            >
               
              <nobr>
                It makes me wonder how I keep from goin' under
              </nobr>
               
            </DropdownItem>
             
            <DropdownItem
              onClick={[Function]}
            >
               
              <nobr>
                -- Grandmaster Flash
              </nobr>
               
            </DropdownItem>
          </Dropdown__DropdownPanel>,
        ],
        "className": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Don't push me...",
              <Dropdown__ToggleIcon
                color="#313233"
                height={9}
                name="caret-down"
                open={false}
                selected={false}
                width={16}
              />,
            ],
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Don't push me...",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "#313233",
                "height": 9,
                "name": "caret-down",
                "open": false,
                "selected": false,
                "width": 16,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              " ",
              <DropdownItem
                onClick={[Function]}
              >
                 
                <nobr>
                  ...cause I'm close to the edge
                </nobr>
                 
              </DropdownItem>,
              " ",
              <DropdownItem
                onClick={[Function]}
              >
                 
                <nobr>
                  I'm trying not to lose my head
                </nobr>
                 
              </DropdownItem>,
              " ",
              <DropdownItem
                onClick={[Function]}
              >
                 
                <nobr>
                  It's like a jungle sometimes
                </nobr>
                 
              </DropdownItem>,
              " ",
              <DropdownItem
                onClick={[Function]}
              >
                 
                <nobr>
                  It makes me wonder how I keep from goin' under
                </nobr>
                 
              </DropdownItem>,
              " ",
              <DropdownItem
                onClick={[Function]}
              >
                 
                <nobr>
                  -- Grandmaster Flash
                </nobr>
                 
              </DropdownItem>,
            ],
            "innerRef": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": Array [
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  " ",
                  <nobr>
                    ...cause I'm close to the edge
                  </nobr>,
                  " ",
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "...cause I'm close to the edge",
                  },
                  "ref": null,
                  "rendered": "...cause I'm close to the edge",
                  "type": "nobr",
                },
                " ",
              ],
              "type": [Function],
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  " ",
                  <nobr>
                    I'm trying not to lose my head
                  </nobr>,
                  " ",
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "I'm trying not to lose my head",
                  },
                  "ref": null,
                  "rendered": "I'm trying not to lose my head",
                  "type": "nobr",
                },
                " ",
              ],
              "type": [Function],
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  " ",
                  <nobr>
                    It's like a jungle sometimes
                  </nobr>,
                  " ",
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "It's like a jungle sometimes",
                  },
                  "ref": null,
                  "rendered": "It's like a jungle sometimes",
                  "type": "nobr",
                },
                " ",
              ],
              "type": [Function],
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  " ",
                  <nobr>
                    It makes me wonder how I keep from goin' under
                  </nobr>,
                  " ",
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "It makes me wonder how I keep from goin' under",
                  },
                  "ref": null,
                  "rendered": "It makes me wonder how I keep from goin' under",
                  "type": "nobr",
                },
                " ",
              ],
              "type": [Function],
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  " ",
                  <nobr>
                    -- Grandmaster Flash
                  </nobr>,
                  " ",
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "-- Grandmaster Flash",
                  },
                  "ref": null,
                  "rendered": "-- Grandmaster Flash",
                  "type": "nobr",
                },
                " ",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Dropdown Component snapshots should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    className={null}
    disabled={false}
    isSelected={false}
    label="Push Me"
  >
     
    <nobr>
      You can place any content you prefer here.
    </nobr>
  </Dropdown>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Dropdown__DropdownButton
          disabled={false}
          onClick={[Function]}
        >
          Push Me
          <Dropdown__ToggleIcon
            color="#313233"
            height={9}
            name="caret-down"
            open={false}
            selected={false}
            width={16}
          />
        </Dropdown__DropdownButton>,
        <Dropdown__DropdownPanel
          innerRef={[Function]}
          open={false}
        >
           
          <nobr>
            You can place any content you prefer here.
          </nobr>
        </Dropdown__DropdownPanel>,
      ],
      "className": null,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Push Me",
            <Dropdown__ToggleIcon
              color="#313233"
              height={9}
              name="caret-down"
              open={false}
              selected={false}
              width={16}
            />,
          ],
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Push Me",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "#313233",
              "height": 9,
              "name": "caret-down",
              "open": false,
              "selected": false,
              "width": 16,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            " ",
            <nobr>
              You can place any content you prefer here.
            </nobr>,
          ],
          "innerRef": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": Array [
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You can place any content you prefer here.",
            },
            "ref": null,
            "rendered": "You can place any content you prefer here.",
            "type": "nobr",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Dropdown__DropdownButton
            disabled={false}
            onClick={[Function]}
          >
            Push Me
            <Dropdown__ToggleIcon
              color="#313233"
              height={9}
              name="caret-down"
              open={false}
              selected={false}
              width={16}
            />
          </Dropdown__DropdownButton>,
          <Dropdown__DropdownPanel
            innerRef={[Function]}
            open={false}
          >
             
            <nobr>
              You can place any content you prefer here.
            </nobr>
          </Dropdown__DropdownPanel>,
        ],
        "className": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Push Me",
              <Dropdown__ToggleIcon
                color="#313233"
                height={9}
                name="caret-down"
                open={false}
                selected={false}
                width={16}
              />,
            ],
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Push Me",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "#313233",
                "height": 9,
                "name": "caret-down",
                "open": false,
                "selected": false,
                "width": 16,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              " ",
              <nobr>
                You can place any content you prefer here.
              </nobr>,
            ],
            "innerRef": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": Array [
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "You can place any content you prefer here.",
              },
              "ref": null,
              "rendered": "You can place any content you prefer here.",
              "type": "nobr",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Dropdown Component snapshots should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    className={null}
    disabled={false}
    isSelected={true}
    label="Push Me"
  >
     
    <DropdownItem
      onClick={[Function]}
    >
       Check that checkmark 
    </DropdownItem>
  </Dropdown>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Dropdown__DropdownButton
          disabled={false}
          onClick={[Function]}
        >
          Push Me
          <Dropdown__ToggleIcon
            color="#89ac52"
            height={9}
            name="caret-selected"
            open={false}
            selected={true}
            width={11}
          />
        </Dropdown__DropdownButton>,
        <Dropdown__DropdownPanel
          innerRef={[Function]}
          open={false}
        >
           
          <DropdownItem
            onClick={[Function]}
          >
             Check that checkmark 
          </DropdownItem>
        </Dropdown__DropdownPanel>,
      ],
      "className": null,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Push Me",
            <Dropdown__ToggleIcon
              color="#89ac52"
              height={9}
              name="caret-selected"
              open={false}
              selected={true}
              width={11}
            />,
          ],
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Push Me",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "#89ac52",
              "height": 9,
              "name": "caret-selected",
              "open": false,
              "selected": true,
              "width": 11,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            " ",
            <DropdownItem
              onClick={[Function]}
            >
               Check that checkmark 
            </DropdownItem>,
          ],
          "innerRef": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": Array [
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": " Check that checkmark ",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": " Check that checkmark ",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Dropdown__DropdownButton
            disabled={false}
            onClick={[Function]}
          >
            Push Me
            <Dropdown__ToggleIcon
              color="#89ac52"
              height={9}
              name="caret-selected"
              open={false}
              selected={true}
              width={11}
            />
          </Dropdown__DropdownButton>,
          <Dropdown__DropdownPanel
            innerRef={[Function]}
            open={false}
          >
             
            <DropdownItem
              onClick={[Function]}
            >
               Check that checkmark 
            </DropdownItem>
          </Dropdown__DropdownPanel>,
        ],
        "className": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Push Me",
              <Dropdown__ToggleIcon
                color="#89ac52"
                height={9}
                name="caret-selected"
                open={false}
                selected={true}
                width={11}
              />,
            ],
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Push Me",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "#89ac52",
                "height": 9,
                "name": "caret-selected",
                "open": false,
                "selected": true,
                "width": 11,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              " ",
              <DropdownItem
                onClick={[Function]}
              >
                 Check that checkmark 
              </DropdownItem>,
            ],
            "innerRef": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": Array [
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": " Check that checkmark ",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": " Check that checkmark ",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
